os/mm/umm_heap/umm_malloc.c

```diff
@@ -229,16 +222,12 @@ FAR void *malloc(size_t size)
 	}
 
 #ifdef CONFIG_DEBUG_MM_HEAPINFO
-	mmaddress_t caller_retaddr = GET_RETURN_ADDRESS();
+	caller_retaddr = GET_RETURN_ADDRESS();
```

Comment:
This should be the same in both enabled and disabled CONFIG_BUILD_KERNEL.
Yes, original code does not have it in enabled. But it's wrong.
Let's set this at  line 190.